cardano-db/src/Cardano/Db/Migration/Extra/CosnumedTxOut/Queries.hs

```diff
@@ -202,40 +222,114 @@ countConsumed = do
     pure countRows
   pure $ maybe 0 unValue (listToMaybe res)
 
-deleteConsumedTxOut :: forall m. MonadIO m => Trace IO Text -> Word64 -> ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut ::
+  forall m.
+  (MonadIO m, MonadBaseControl IO m) =>
+  Trace IO Text ->
+  Word64 ->
+  ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut trce blockNoDiff = do
+  maxTxInId <- findMaxTxInId blockNoDiff
+  case maxTxInId of
+    Left _ -> do
+      createConsumedTxOut
+      liftIO $ logInfo trce "Created ConsumedTxOut as there were no blocks present"
```

Comment:
The comment here is a bit misleading. The returned Left message, describes in a better way why no entries need to be deleted.

Other than that, this case should still call `splitAndProcessPageEntries` and update for all `tx_in` entries.

Reply:
it can't as `migrateNextPage` needs `mxtxid` might need a code example of what you are thinking.

Reply:
It could take a `Maybe` and only update if it's `Nothing`

Reply:
it can't as `migrateNextPage` needs `mxtxid` might need a code example of what you are thinking.
Also the call site used to be:
```
(False, _, True) -> do
          shouldInsertToMigrationTable
          liftIO $ logInfo trce "Running extra migrations consumed_tx_out and prune tx_out"
          ExtraCons.migrateTxOut (Just trce)
          liftIO $ logInfo trce "Now Running extra migration prune tx_out"
          ExtraCons.deleteConsumedTxOut trce blockNoDiff
         ...
```
`deleteConsumedTxOut` just logs for our left:
```
deleteConsumedTxOut trce blockNoDiff = do
  maxTxInId <- findMaxTxInId blockNoDiff
  case maxTxInId of
    Left errMsg -> liftIO $ logInfo trce $ "No tx_out was deleted: " <> errMsg
```
so to be the same as before `deleteAndUpdateConsumedTxOut` needs to `migrateTxOut` + `logInfo`, unless I'm missing something?

Reply:
Yes so the current implementation doesn't log the Left result of `findMaxTxInId` which I think it should.

Reply:
ah ok I didn't understand this part:
> this case should still call `migrateNextPage` and update for all `tx_in` entries
I'll change the log wording 👍

Reply:
These are two separate issues actually.